crucible:helper-tests fail to compile with tasty-hspec-1.1.7

`crucible`'s `helper-tests` fail to compile with the latest version of `tasty-hspec`:

```
$ cabal test crucible:helper-tests --constraint "tasty-hspec==1.1.7"
Build profile: -w ghc-8.10.4 -O1
In order, the following will be built (use -v for more details):
 - crucible-0.5 (test:helper-tests) (first run)
Preprocessing test suite 'helper-tests' for crucible-0.5..
Building test suite 'helper-tests' for crucible-0.5..
[1 of 1] Compiling Main             ( test/helpers/Main.hs, /home/rscott/Documents/Hacking/Haskell/sandbox/crucible/dist-newstyle/build/x86_64-linux/ghc-8.10.4/crucible-0.5/t/helper-tests/build/helper-tests/helper-tests-tmp/Main.o ) [Test.Tasty.Hspec changed]

test/helpers/Main.hs:20:11: error:
    Variable not in scope:
      describe
        :: [Char] -> t0 -> hspec-core-2.8.1:Test.Hspec.Core.Spec.Monad.Spec
   |
20 |           describe "panic" $
   |           ^^^^^^^^

test/helpers/Main.hs:21:11: error:
    • Variable not in scope: it :: [Char] -> t2 -> t0
    • Perhaps you meant ‘id’ (imported from Prelude)
   |
21 |           it "should throw an exception with the right details" $
   |           ^^

test/helpers/Main.hs:22:11: error:
    Variable not in scope:
      shouldThrow :: t1 -> (Panic Crucible -> Bool) -> t2
   |
22 |           shouldThrow (panic "Oh no!" ["line 1", "line 2"]) acceptableExn
   |           ^^^^^^^^^^^
```

On the other hand, they compile with earlier `tasty-hspec` releases, as `cabal test test:helper-tests --constraint "tasty-hspec==1.1.6"` works as you'd expect.

This happens because `tasty-hspec-1.1.7` no longer re-exports `Test.Hspec`. (These changes probably ought to have come with a major version bump, but oh well.) In any case, the fix is straightforward, so I'll submit it soon.

--- a/crucible/crucible.cabal
+++ b/crucible/crucible.cabal
@@ -157,6 +157,7 @@ test-suite helper-tests
   main-is: Main.hs
   build-depends: base,
                  containers,
+                 hspec >= 2.5,
                  mtl,
                  crucible,
                  what4,

--- a/crucible/test/helpers/Main.hs
+++ b/crucible/test/helpers/Main.hs
@@ -3,8 +3,9 @@ module Main where
 
 import Data.List (isInfixOf)
 
-import qualified Test.Tasty as T
-import Test.Tasty.Hspec
+import Test.Hspec
+import Test.Tasty
+import Test.Tasty.Hspec (testSpec)
 
 import Lang.Crucible.Panic
 
@@ -12,15 +13,15 @@ import qualified Panic as P
 
 main :: IO ()
 main =
-  T.defaultMain =<< panicTests
+  defaultMain =<< panicTests
 
-panicTests :: IO T.TestTree
+panicTests :: IO TestTree
 panicTests =
   do t <- testSpec "Panicking throws an exception" $
           describe "panic" $
           it "should throw an exception with the right details" $
           shouldThrow (panic "Oh no!" ["line 1", "line 2"]) acceptableExn
-     pure $ T.testGroup "panic" [ t ]
+     pure $ testGroup "panic" [ t ]
   where
     acceptableExn :: P.Panic Crucible -> Bool
     acceptableExn e =

